locator/token_metadata.hh

```diff
@@ -58,6 +58,10 @@ struct host_id_or_endpoint {
 
     host_id_or_endpoint(const sstring& s, param_type restrict = param_type::auto_detect);
 
+    bool operator==(const host_id_or_endpoint& o) const noexcept {
+        return has_host_id() ? id == o.id : endpoint == o.endpoint;
```

Comment:
If both host id and endpoint can be set then this operator is not symmetric, which can lead to undefined behavior.

Example:

a = {null, 10.0.0.1}
b = {id1, 10.0.0.1}

a == b is true
b == a is false

Reply:
We should document that "or" in the name is not an exclusive "or", it's not obvious.

Reply:
maybe and_or would be better then, though it's more cumbersome

Reply:
> If both host id and endpoint can be set then this operator is not symmetric, which can lead to undefined behavior.

Will fix

Reply:
fixed and renamed in v7